files/amq_metrics

```diff
@@ -70,49 +109,50 @@ end
 
 filename = Time.now.utc.strftime('%Y%m%dT%H%M%SZ') + '.json'
 
-HOSTS.each do |host|
+config[:hosts].each do |host|
   begin
     timestamp = Time.now
     dataset = {'timestamp' => timestamp.utc.iso8601, 'servers' => {}}
     hostkey = host.gsub('.', '-')
-    dataset['servers'][hostkey] = {METRICS_TYPE => {}}
+    dataset['servers'][hostkey] = {config[:"metrics-type"] => {}}
 
-    host_url = "https://#{host}:#{PORT}/api/jolokia"
-    response = get_endpoint(host_url, POST_DATA)
+    host_url = "http://#{host}:#{config[:"metrics-port"]}/api/jolokia"
+    response = get_endpoint(host_url, config[:"additional-metrics"].to_json)
 
     JSON.parse(response.body).each do |element|
       case element['value']
       when Hash
         element['value'].each do |mbean,attributes|
           metrics = treeify(mbean, attributes)
-          recurse_merge!(dataset['servers'][hostkey][METRICS_TYPE], metrics)
+          recurse_merge!(dataset['servers'][hostkey][config[:"metrics-type"]], metrics)
         end
       else
         req = element['request']
         metrics = treeify(req['mbean'], { req['attribute'] => req['value'] })
-        recurse_merge!(dataset['servers'][hostkey][METRICS_TYPE], metrics)
+        recurse_merge!(dataset['servers'][hostkey][config[:"metrics-type"]], metrics)
       end
     end
 
-    dataset['servers'][hostkey][METRICS_TYPE]['error'] = $error_array
-    dataset['servers'][hostkey][METRICS_TYPE]['error_count'] = $error_array.count
-    dataset['servers'][hostkey][METRICS_TYPE]['api-query-start'] = timestamp.utc.iso8601
-    dataset['servers'][hostkey][METRICS_TYPE]['api-query-duration'] = Time.now - timestamp
+    dataset['servers'][hostkey][config[:"metrics-type"]]['error'] = $error_array
+    dataset['servers'][hostkey][config[:"metrics-type"]]['error_count'] = $error_array.count
+    dataset['servers'][hostkey][config[:"metrics-type"]]['api-query-start'] = timestamp.utc.iso8601
+    dataset['servers'][hostkey][config[:"metrics-type"]]['api-query-duration'] = Time.now - timestamp
 
     json_dataset = JSON.pretty_generate(dataset)
 
-    unless OUTPUT_DIR.nil? then
-      Dir.chdir(OUTPUT_DIR) do
+    if config[:"output-dir"] then
+      Dir.chdir(config[:"output-dir"]) do
         Dir.mkdir(host) unless File.exist?(host)
         File.open(File.join(host, filename), 'w') do |file|
           file.write(json_dataset)
         end
       end
     end
-    if options[:print] != false then
+    if config[:print] then
```

Comment:
The default behavior of the script is currently to print.  I think this breaks that unless you've defaulted print to true somehow.  Will test.